catalyst_voices/apps/voices/integration_test/pageobject/spaces_drawer_page.dart

```diff
@@ -0,0 +1,38 @@
+library spaces_drawer_page;
+
+import 'package:catalyst_voices_models/catalyst_voices_models.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+class SpacesDrawerPage {
+  static final closeBtn = find.byKey(const Key('MenuCloseButton'));
+  static final guestMenuItems = find.byKey(const Key('GuestMenuItems'));
+  static final allSpacesBtn =
+      find.byKey(const Key('DrawerChooserAllSpacesButton'));
+  static final chooserPrevBtn =
+      find.byKey(const Key('DrawerChooserPreviousButton'));
+  static final chooserNextBtn =
+      find.byKey(const Key('DrawerChooserNextButton'));
+  static final chooserItemContainer =
+      find.byKey(const Key('DrawerChooserItem'));
+
+  static Finder Function(Space space) get chooserItem => (Space space) {
+        return find.byKey(Key('DrawerChooser$space'));
+      };
+
+  static Finder Function(Space name) get chooserIcon => (Space space) {
+        return find.byKey(Key('DrawerChooser${space}AvatarKey'));
+      };
+
+  static void Function() get looksAsExpected => () {
+        expect(closeBtn, findsOneWidget);
+        expect(allSpacesBtn, findsOneWidget);
+        expect(chooserPrevBtn, findsOneWidget);
+        expect(chooserNextBtn, findsOneWidget);
+        expect(chooserItemContainer, findsExactly(5));
+        expect(
+          chooserIcon(Space.discovery),
+          findsOneWidget,
+        );
+      };
```

Comment:
Drawer will look differently depending on the session state, roles, admin status, etc, so most likely this function will need to envolve and you'll end up with different variants per role, etc.

Reply:
Yep I'm aware, it's about head start, having as much as we can in a current state. Getting common technical issues solved in the process.